rust/gitxetcore/src/git_integration/git_repo.rs

```diff
@@ -1582,7 +1661,7 @@ impl GitRepo {
         }
 
         // Only need to install guard notes when this is an upgrade.
-        if self.mdb_version < *version {
+        if self.mdb_version < *version && self.mdb_version != ShardVersion::Unitialized {
```

Comment:
Why skip setting guard notes if it's uninitialized? Although this is not an issue when no one uses old clients.

Reply:
In this case, this function gets called from many different places.  Given that this functionality is only needed when cloning from a github repo (currently), I want to isolate it to that case.  
When first creating a repo, or when running git in a non-xet repo, the notes may not be available yet and I don't want there to be unintended side effects when running e.g. status or other commands.  At least with the filter, we can be confident that it's only being run in a xet repo.

Reply:
I see, so we postpone setting the guard from init to the filter process.